File: BHOI_/2014/BHOI 2014/Piramida/grader.cpp
#include <iostream>
#include <cassert>

using namespace std;

int MaxUcinak(int n, int k, int p, int s, int* poeni);

int main() {
  int n, k, p, s;
  cin >> n >> k >> p >> s;
  int* poeni = new int [n * k];
  for (int i = 0; i < n * k; i++)
      cin >> poeni[i];
  cout << MaxUcinak(n, k, p, s, poeni) << endl;
  return 0;
}

File: BHOI_/2014/BHOI 2014/Piramida/grader.pas
Program grader;
uses piramida;
const
   inbuf_len  = 1 shl 23;
   outbuf_len = 1 shl 10;
var
  n, k, p, s, i : LongInt;
  poeni : Array of LongInt;
  inbuf : array [1..inbuf_len] of Byte;
  outbuf : array [1..outbuf_len] of Byte;
begin
  settextbuf(input, inbuf);
  settextbuf(output, outbuf);
  readln(n, k, p, s);
  SetLength(poeni, n * k);
  for i := 0 to (n * k) - 1 do
    read(poeni[i]);
  writeln(MaxUcinak(n, k, p, s, poeni));
end.
